pkg/federation/sdlmerge/remove_scalar_duplicates.go

```diff
@@ -0,0 +1,52 @@
+package sdlmerge
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+type removeDuplicateScalarTypeDefinitionVisitor struct {
+	operation     *ast.Document
+	scalarSet     map[string]bool
+	nodesToRemove []ast.Node
+	lastRef       int
+}
+
+func newRemoveDuplicateScalarTypeDefinitionVistior() *removeDuplicateScalarTypeDefinitionVisitor {
+	return &removeDuplicateScalarTypeDefinitionVisitor{
+		nil,
+		make(map[string]bool),
+		make([]ast.Node, 0),
+		-1,
```

Comment:
we have a constant for this magic number: 
https://github.com/TykTechnologies/graphql-go-tools/blob/master/pkg/ast/ast.go#L8